ide/html.editor/src/org/netbeans/modules/html/editor/hints/other/AddMissingAltAttributeRule.java

```diff
@@ -0,0 +1,234 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.netbeans.modules.html.editor.hints.other;
+
+import java.util.LinkedList;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicReference;
+import javax.swing.text.BadLocationException;
+import javax.swing.text.Document;
+import org.netbeans.api.html.lexer.HTMLTokenId;
+import org.netbeans.api.lexer.Token;
+import org.netbeans.api.lexer.TokenHierarchy;
+import org.netbeans.api.lexer.TokenSequence;
+import org.netbeans.api.project.FileOwnerQuery;
+import org.netbeans.api.project.Project;
+import org.netbeans.modules.csl.api.Hint;
+import org.netbeans.modules.csl.api.HintSeverity;
+import org.netbeans.modules.csl.api.OffsetRange;
+import org.netbeans.modules.csl.api.RuleContext;
+import org.netbeans.modules.html.editor.api.Utils;
+import org.netbeans.modules.html.editor.api.gsf.HtmlParserResult;
+import org.netbeans.modules.html.editor.hints.HtmlRule;
+import org.netbeans.modules.html.editor.hints.HtmlRuleContext;
+import org.netbeans.modules.html.editor.refactoring.InlinedStyleInfo;
+import org.netbeans.modules.html.editor.utils.HtmlTagContextUtils;
+import org.netbeans.modules.web.common.api.WebUtils;
+import org.openide.filesystems.FileObject;
+import org.openide.util.Exceptions;
+import org.openide.util.NbBundle;
+
+/**
+ *
+ * @author Chris
+ */
+@NbBundle.Messages("AddMissingAltAttributeRuleName=Insert required 'alt' attribute")
+public class AddMissingAltAttributeRule extends HtmlRule {
+
+    public static AddMissingAltAttributeRule SINGLETON = new AddMissingAltAttributeRule();
+    private RuleContext context;
+
+    @Override
+    public boolean appliesTo(RuleContext context) {
+        this.context = context;
+
+        HtmlParserResult result = (HtmlParserResult) context.parserResult;
+        FileObject file = result.getSnapshot().getSource().getFileObject();
+
+        if (file == null) {
+            return false;
+        }
+
+        Project project = FileOwnerQuery.getOwner(file);
+
+        if (project == null) {
+            return false;
+        }
```

Comment:
Why is the project checked - it is not used later. This results in this:

![image](https://user-images.githubusercontent.com/2179736/183995129-691d1d7c-1e6e-4c41-872d-052cec39e84a.png)

The cursor is in the img-Tag (see my general comment), I get the error indicator, but I don't get the fix.

Reply:
Thx for the hint, yeah I didn't test an HTML file w/o having a project :).